src/planner/dart/ConfigurationToConfiguration_to_ConfigurationToConfigurations.cpp

```diff
@@ -0,0 +1,85 @@
+#include "aikido/planner/dart/ConfigurationToConfiguration_to_ConfigurationToConfigurations.hpp"
+
+#include <dart/dynamics/dynamics.hpp>
+
+#include "aikido/common/RNG.hpp"
+#include "aikido/distance/NominalConfigurationRanker.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSaver.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+using aikido::distance::ConstConfigurationRankerPtr;
+using aikido::distance::NominalConfigurationRanker;
+using aikido::statespace::dart::MetaSkeletonStateSaver;
+using aikido::statespace::dart::MetaSkeletonStateSpace;
+
+namespace aikido {
+namespace planner {
+namespace dart {
+
+//==============================================================================
+ConfigurationToConfiguration_to_ConfigurationToConfigurations::
+    ConfigurationToConfiguration_to_ConfigurationToConfigurations(
+        std::shared_ptr<planner::ConfigurationToConfigurationPlanner> planner,
+        ::dart::dynamics::MetaSkeletonPtr metaSkeleton,
+        distance::ConstConfigurationRankerPtr configurationRanker)
+  : PlannerAdapter<
+        planner::ConfigurationToConfigurationPlanner,
+        ConfigurationToConfigurationsPlanner>(
+        std::move(planner), std::move(metaSkeleton))
+{
+  mConfigurationRanker = std::move(configurationRanker);
+}
+
+//==============================================================================
+trajectory::TrajectoryPtr
+ConfigurationToConfiguration_to_ConfigurationToConfigurations::plan(
+    const ConfigurationToConfigurations& problem, Planner::Result* result)
+{
+  // TODO: Check equality between state space of this planner and given problem.
+
+  auto saver = MetaSkeletonStateSaver(mMetaSkeleton);
+  DART_UNUSED(saver);
+
+  // Get the start state from the MetaSkeleton, since this is a DART planner.
+  auto startState = mMetaSkeletonStateSpace->createState();
+  mMetaSkeletonStateSpace->getState(mMetaSkeleton.get(), startState);
+
+  // Use a ranker
+  ConstConfigurationRankerPtr configurationRanker(mConfigurationRanker);
+  if (!configurationRanker)
+  {
+    auto nominalState = mMetaSkeletonStateSpace->createState();
+    mMetaSkeletonStateSpace->copyState(startState, nominalState);
+    configurationRanker = std::make_shared<const NominalConfigurationRanker>(
+        mMetaSkeletonStateSpace, mMetaSkeleton, nominalState);
+  }
+
+  std::vector<MetaSkeletonStateSpace::ScopedState> configurations;
+  for (const auto& goal : problem.getGoalStates())
```

Comment:
Nit: Is there a reason to use a ref here? I think each `goal` is just a pointer/adress.